pkgs/os-specific/linux/zfs/default.nix

```diff
@@ -26,6 +26,8 @@ let
   buildKernel = any (n: n == configFile) [ "kernel" "all" ];
   buildUser = any (n: n == configFile) [ "user" "all" ];
 
+  stdenv' = if kernel == null then stdenv else kernel.stdenv;
```

Comment:
Maybe add a comment what this fixes as it is non-obvious.

Reply:
Done :)